[Triton] [Inductor] Enable Epilogue Subtiling in the blackwell ws template

njriasan opened the pull request:
Summary: Enables support for epilogue subtiling in the blackwell ws template. This requires the ability to call `store_output` twice in the same kernel and reuse the same tensor descriptor across allocations.

Test Plan:
Tested with test_max_autotune.py on a Blackwell server.

Rollback Plan:

Differential Revision: D82610077




cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @chenyang78 @kadeng @muchulee8 @amjames @chauhang @aakhundov @coconutruben @mlazos

njriasan commented on test/inductor/test_max_autotune.py:
```diff
@@ -271,12 +271,14 @@ def next_multiple_16(a: int) -> int:
     @parametrize("b_transposed", (False, True))
     @parametrize("dynamic", (False, True))
     @parametrize("tma_store", (False, True))
+    @parametrize("epilogue_subtile", (False, True))
```
At this time I was checking that the code doesn't crash, but I should just look for the store count (2 vs 1). I can update the test. Thanks for the feedback.

PaulZhang12 commented on torch/_inductor/codegen/triton.py:
```diff
@@ -2485,42 +2487,49 @@ def codegen_block_ptr(
-                self.block_ptr_to_buffer[block_descriptor] = name
+                block_descriptor_id = next(self.block_ptr_id)
+                if isinstance(indexing, BlockPtrOptions):
+                    block_descriptor = f"block_ptr{block_descriptor_id}"
```
Did you test with block pointer? I thought we don't really use block_ptr anymore in triton

njriasan committed:
[Triton] [Inductor] Enable Epilogue Subtiling in the blackwell ws template (#163145)

Summary:

Enables supprot for epilogue subtiling in the blackwell ws template. This requires the ability to call `store_output` twice in the same kernel and reuse the same tensor descriptor across allocations.

Test Plan: Tested with test_max_autotune.py on a Blackwell server.

Differential Revision: D82610077

eellison reviewed (commented):
looks good ! would it be possible to test this with @drisspg's recent b200 enablement ?

njriasan committed:
Applied remaining feedback

njriasan committed:
Fixed linting

njriasan committed:
Fixed test failures

njriasan committed:
Removed lintrunner error

njriasan commented:
@pytorchbot merge
